internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -198,6 +212,64 @@ func (e *EnvStackConfig) StackName() string {
 	return NameForEnv(e.in.App.Name, e.in.Name)
 }
 
+// transformParameters removes or transforms each of the current parameters and does not add any new parameters.
```

Comment:
this section is an exact copy of https://github.com/aws/copilot-cli/blob/41856491bfda15f4eec1df7372d07c65a181b89d/internal/pkg/deploy/cloudformation/env.go#L262-L315 

We'll send follow-up PRs after this to deprecate "env upgrade" all that code will be deleted

Reply:
Yup that's the plan, as discussed here: https://github.com/aws/copilot-cli/pull/3750#discussion_r919444192 `UpdateAndRenderEnvironment` can invoke a slightly different constructor that will leverage `UsePreviousValue: true` just like today